lib/designator/subject_set_cache.ex

```diff
@@ -61,7 +61,13 @@ defmodule Designator.SubjectSetCache do
     ConCache.put(:subject_set_cache, key, subject_set)
   end
 
-  def reload(key) do
+  def reload({workflow_id, nil} = key) do
```

Comment:
Similarly it looks like there are 2 `reload` methods, it's been a while since I've done elixir, I'm assuming if `subject_set_id` coming from `workflow_controller.ex` line to 63 is `nil` then `subject_set_cache.ex`'s `reload` on line 64 is hit if not then the `reload` in line 70 is hit? 

Also following GH action comment, it looks like both `workflow_id` (and `subject_set_id` for `reload` method in line 70) is not used, would it be better for readability to prefix with underscore?

Reply:
> Similarly it looks like there are 2 `reload` methods, it's been a while since I've done elixir, I'm assuming if `subject_set_id` coming from `workflow_controller.ex` line to 63 is `nil` then `subject_set_cache.ex`'s `reload` on line 64 is hit if not then the `reload` in line 70 is hit?

Nvm. Forgot a big part of elixir which is overloading. So to answer my own question, yes, it will do just that.

Reply:
yes - elixir offers pattern matching behaviours on function arity, types and values. This allow us to easily extract out behaviour to similarly names functions without gaurd clauses etc. 

TBH i quite like this as it allow one to simplify the function logic and and simplify the call site.